Generated guest code with wrong lifetime argument

example.wit:

```wit
interface http-fetch-imports {
    record request{
        method: string,
        uri: string,
        body: string
    }

    record response{
        status: u16,
        body: string
    }

    fetch: func(req: request) -> result<response>
}

default world http-fetch-simple {
    import http-fetch-simple: self.http-fetch-imports
}
```

generate guest code:

```bash
wit-bindgen rust example.wit
```

error message in code:

```rust
// Generated by `wit-bindgen` 0.3.0. DO NOT EDIT!

#[allow(clippy::all)]
pub mod http_fetch_simple{
  #[derive(Clone)]
  pub struct Response {
    pub status: u16,
    pub body: wit_bindgen::rt::string::String,
  }
  impl core::fmt::Debug for Response {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
      f.debug_struct("Response").field("status", &self.status).field("body", &self.body).finish()
    }
  }
  #[derive(Clone)]
  pub struct Request {
    pub method: wit_bindgen::rt::string::String,
    pub uri: wit_bindgen::rt::string::String,
    pub body: wit_bindgen::rt::string::String,
  }
  impl core::fmt::Debug for Request {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
      f.debug_struct("Request").field("method", &self.method).field("uri", &self.uri).field("body", &self.body).finish()
    }
  }
  #[allow(clippy::all)]
  // this line show error[E0107]: this struct takes 0 lifetime arguments but 1 lifetime argument was supplied
  pub fn fetch(req: Request<'_,>,) -> Result<Response,()>{ 
  ......
}
```

--- a/crates/gen-rust-lib/src/lib.rs
+++ b/crates/gen-rust-lib/src/lib.rs
@@ -382,7 +382,13 @@ pub trait RustGenerator<'a> {
     fn modes_of(&self, ty: TypeId) -> Vec<(String, TypeMode)> {
         let info = self.info(ty);
         let mut result = Vec::new();
-        result.push((self.result_name(ty), TypeMode::Owned));
+        let first_mode = if info.owned || !info.borrowed {
+            TypeMode::Owned
+        } else {
+            assert!(!self.uses_two_names(&info));
+            TypeMode::AllBorrowed("'a")
+        };
+        result.push((self.result_name(ty), first_mode));
         if self.uses_two_names(&info) {
             result.push((self.param_name(ty), TypeMode::AllBorrowed("'a")));
         }

--- a/tests/codegen/simple-http.wit
+++ b/tests/codegen/simple-http.wit
@@ -1,0 +1,18 @@
+interface http-fetch-imports {
+    record request{
+        method: string,
+        uri: string,
+        body: string
+    }
+
+    record response{
+        status: u16,
+        body: string
+    }
+
+    fetch: func(req: request) -> result<response>
+}
+
+default world http-fetch-simple {
+    import http-fetch-simple: self.http-fetch-imports
+}

